internal/modelproxy/request.go

```diff
@@ -38,14 +38,15 @@ func newProxyRequest(r *http.Request) *proxyRequest {
 	}
 
 	return pr
-
 }
 
-// parseModel attempts to determine the model from the request.
+// parse attempts to determine the model from the request.
 // It first checks the "X-Model" header, and if that is not set, it
 // attempts to unmarshal the request body as JSON and extract the
 // .model field.
-func (pr *proxyRequest) parseModel() error {
+func (pr *proxyRequest) parse() error {
+	pr.selectors = pr.r.Header.Values("X-Selector")
```

Comment:
if you want to fail fast, you could to some basic validation on the selector. But it is ok to do this later.

Reply:
keeping as fmt.Print b/c its hard to read logged yaml

Reply:
will keep label parsing logic to live in the pkg that deal with k8s stuff